### PR TITLE
[WIP] fix: l2geth failing tests

### PR DESCRIPTION
**Pull request by NazariiDenha**
## 1. Purpose or design rationale of this PR
Find and fix failing tests `l2geth`

https://www.notion.so/scrollzkp/l2geth-failing-tests-f5f8b38b2d4047a3b4f55e01770d3455?pvs=4

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [x] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [ ] This PR is not a breaking change
- [ ] Yes

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205321339560887

### CODE REVIEW

File: consensus/misc/eip1559.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
 		return err
 	}
 	// Verify the header is not malformed
-	if header.BaseFee == nil && config.Scroll.BaseFeeEnabled() {
+	if (header.BaseFee == nil && config.Scroll.BaseFeeEnabled()) || (header.BaseFee != nil && !config.Scroll.BaseFeeEnabled()) {
```
Comment (Thegaram) [inline consensus/misc/eip1559.go]: Need a different error message for the new case

File: eth/gasprice/gasprice_test.go [hunk 2]
```diff
@@ -108,6 +108,7 @@ func newTestBackend(t *testing.T, londonBlock *big.Int, pending bool) *testBacke
 	)
 	config.LondonBlock = londonBlock
 	config.ArrowGlacierBlock = londonBlock
+	config.ArchimedesBlock = londonBlock
```
Comment (Thegaram) [inline eth/gasprice/gasprice_test.go:111]: Do we need `config.ShanghaiBlock`?

File: eth/tracers/js/tracer_test.go [hunk 3]
```diff
@@ -227,37 +227,6 @@ func TestIsPrecompile(t *testing.T) {
 		t.Errorf("Tracer should consider blake2f as precompile in istanbul")
 	}
 
-	// test sha disabled in archimedes
```
Comment (Thegaram) [inline eth/tracers/js/tracer_test.go:230]: Why do we need to remove these?

Comment (NazariiDenha) [reply eth/tracers/js/tracer_test.go:230]: there is info about it in [notion page](https://www.notion.so/scrollzkp/l2geth-failing-tests-f5f8b38b2d4047a3b4f55e01770d3455?pvs=4#066c13f87fa4488a8e59882e75f0e2f4)

File: eth/gasprice/gasprice_test.go (hunk 2 above)
Comment (NazariiDenha) [reply eth/gasprice/gasprice_test.go:111]: no, those tests don't test anything regarding `Shanghai` just fails on [CheckConfigForkOrder](https://github.com/scroll-tech/go-ethereum/blob/805676e22721093515332352eee02ff64b6d8caf/params/config.go#L724)

File: consensus/misc/eip1559.go (hunk 1 above)
Comment (NazariiDenha) [reply consensus/misc/eip1559.go]: updated

**Comment by Thegaram** [comment]
I've seen some tests fail non-deterministically, including these:

```
TestGolangBindings
TestOverload
TestWalletNotifications
TestOversizedTxThenNormal
```

**Pull request closed without merging**